python/graphstorm/inference/ep_infer.py

```diff
@@ -119,10 +124,20 @@ def infer(self, loader, save_embed_path, save_prediction_path=None,
 
             # The order of the ntypes must be sorted
             embs = {ntype: embs[ntype] for ntype in sorted(target_ntypes)}
-            save_gsgnn_embeddings(save_embed_path, embs, self.rank,
-                get_world_size(),
-                device=device,
-                node_id_mapping_file=node_id_mapping_file)
+
+            if not saved_hdf5_embed:
+                save_gsgnn_embeddings(save_embed_path, embs, self.rank,
```

Comment:
You can change the implementation of save_gsgnn_embeddings instead of change all xx_infer.py

Reply:
By passing save_embed_format, you can let save_gsgnn_embeddings decide which data format will be used.

Reply:
Actually I feel gnn inference doesn't need to differentiate lp, ep or np cuz these three are training objectives and should have nothing to do with inference. Can we optimize this by unify gnn inference in future?

Reply:
updated as suggested